File: net/JetBrains.Serialization/src/Dataholders/IssuerSerialInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using JetBrains.Serialization;
using Org.BouncyCastle.Asn1.X509;

[JsonObject(MemberSerialization.OptIn)]
public class IssuerSerialInfo : IEncodableInfo
{
  [JsonProperty("GeneralNames")] private List<GeneralNameInfo> _generalNames;
  [JsonProperty("Serial")] private TextualInfo _serial;
  [JsonProperty("IssuerUID")] private TextualInfo? _issuerUid;

  [JsonConstructor]
  public IssuerSerialInfo(List<GeneralNameInfo> generalNames, TextualInfo serial, TextualInfo? issuerUid)
  {
    _generalNames = generalNames;
    _serial = serial;
    _issuerUid = issuerUid;
  }

  public IssuerSerialInfo(IssuerSerial issuer)
  {
    _generalNames = issuer.Issuer.GetNames().Select(name => new GeneralNameInfo(name)).ToList();
    _serial = TextualInfo.GetInstance(issuer.Serial);
    _issuerUid = issuer.IssuerUid != null ? TextualInfo.GetInstance(issuer.IssuerUid) : null;
  }

  public Asn1Encodable ToPrimitive()
  {
    var asn1Items = new List<Asn1Encodable>
    {
      _generalNames.ToPrimitiveDerSequence(),
      _serial.ToPrimitive()
    };

    if (_issuerUid != null)
    {
      asn1Items.Add(_issuerUid.ToPrimitive());
    }

    return asn1Items.ToDerSequence();
  }
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/IssuerSerialInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.BigIntegerSerializer
import com.jetbrains.signatureverifier.serialization.toDLSequence
import com.jetbrains.signatureverifier.serialization.toPrimitiveDLSequence
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.ASN1Integer
import org.bouncycastle.asn1.ASN1Primitive
import org.bouncycastle.asn1.x509.IssuerSerial
import java.math.BigInteger

@Serializable
data class IssuerSerialInfo(
  val generalNames: List<GeneralNameInfo>,
  @Serializable(BigIntegerSerializer::class)
  val serial: BigInteger,
  val issuerUID: TextualInfo?
) : EncodableInfo {

  constructor(issuer: IssuerSerial) : this(
    issuer.issuer.names.map { GeneralNameInfo(it) },
    issuer.serial.value,
    issuer.issuerUID?.let { TextualInfo.getInstance(it) }
  )

  override fun toPrimitive(): ASN1Primitive =
    listOf(
      generalNames.toPrimitiveDLSequence(),
      ASN1Integer(serial),
      issuerUID?.toPrimitive()
    ).toDLSequence()
}